src/ScanPatron/ScanPatron.js

```diff
@@ -1,7 +1,124 @@
-import React from 'react';
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import { get } from 'lodash';
 
-const ScanPatron = () => (
-  <div>ScanPatron</div>
-);
+import {
+  Button,
+  Paneset,
+  Pane,
+  PaneFooter,
+} from '@folio/stripes/components';
+import { stripesConnect } from '@folio/stripes/core';
 
-export default ScanPatron;
+import ScanForm from './ScanForm';
+import PatronDetail from '../PatronDetail';
+
+const ScanPatron = ({ mutator, resources }) => {
+  const isUserProfilePicConfigEnabledForTenant = get(resources, 'userProfilePicConfig.records[0].enabled');
+  const [scannedPatronDetails, setScannedPatronDetails] = useState();
+
+  const handleScanPatron = async (barcode) => {
+    if (barcode) {
+      const query = `barcode==${barcode}`;
+      const patron = await mutator.patrons.GET({ params: { query } });
+
+      if (patron.length) {
```

Comment:
can patron variable be undefined ? Is optional chaining needed here, please have a look.

Reply:
1. added check for patron
2. this block of code will grow and hence I don't think optional chaining will be suitable here.